src/extensions/lens-extension.ts

```diff
@@ -83,59 +83,44 @@ export class LensExtension {
   }
 
   @action
-  async enable() {
-    if (this.isEnabled) return;
-    this.isEnabled = true;
-    this.onActivate?.();
-    logger.info(`[EXTENSION]: enabled ${this.name}@${this.version}`);
+  async enable(register: (ext: LensExtension) => Promise<Disposer[]>) {
```

Comment:
Not directly related to this PR but could we move `enable/disable` methods to `ExtensionLoader`? It feels like they are in a wrong place atm.

Reply:
Probably, see #2245 but I think we can and should do away with the registries as well (non-breaking change)